doc/modules/metrics.rst

```diff
@@ -177,7 +177,7 @@ It can be computed using :func:`chi2_kernel` and then passed to an
 
 It can also be directly used as the ``kernel`` argument::
 
-    >>> svm = SVC(kernel=chi2_kernel).fit(X, y)
+    >>> svm = SVC(gamma='scale', kernel=chi2_kernel).fit(X, y)
```

Comment:
@gxyd also ping @jnothman 
I wonder whether it makes sense to pass gamma='scale' here, See below:
```python
X = [[0, 1], [1, 0], [.2, .8], [.7, .3]]
y = [0, 1, 0, 1]
svm = SVC(gamma=0.5, kernel=chi2_kernel, probability=True).fit(X, y)
test1 = svm.predict_proba(X)
svm = SVC(gamma=1, kernel=chi2_kernel, probability=True).fit(X, y)
test2 = svm.predict_proba(X)
assert_array_almost_equal(test1, test2) # True, gamma is ignored and set to default(1.0)

K = chi2_kernel(X, gamma=0.5)
svm = SVC(kernel='precomputed', probability=True).fit(K, y)
test3 = svm.predict_proba(K)
K = chi2_kernel(X, gamma=1)
svm = SVC(kernel='precomputed', probability=True).fit(K, y)
test4 = svm.predict_proba(K)
assert_array_almost_equal(test1, test3) # False
assert_array_almost_equal(test1, test4) # True
```
In this case, we'll get unnecessary warnings since gamma is ignored? Or maybe it's a bug? (Or sorry if I'm wrong)

Reply:
I don't think this should be considered a bug, since the documentation of gamma says:
```
gamma : float, optional (default=’auto’)

    Kernel coefficient for ‘rbf’, ‘poly’ and ‘sigmoid’. If gamma is ‘auto’ then 1/n_features will be used instead.
```

It doesn't say, that 'gamma' will be used for a 'callable' or 'precompute'. Unless we consider this to be some problem in the documentation, I see no problem with current results of SVC neglecting gamma, when a callable is passed.

So, implying that it isn't meaningful for a callable kernel or for kernel='precompute'.

Reply:
> I don't think this should be considered a bug

Agree. We have other ways (e.g., functools.partial) to specify parameters for a callable. So correct the misleading example and avoid unnecesssary warnings seem enough from my side.

Reply:
Yes, that's how I tested locally as well. I wasn't aware of `functools.partial` until I worked on my other PR (regarding jaccard_similarity_score).